### ISSUE TITLE
W4701(modified-iterating-list) not reported for glob.glob() list

### ISSUE THREAD

**Issue by jakub-krysl-ipf** [pylint-dev/pylint#9315]
### Bug description

```python
"""This shows missing W4701"""
import glob

files = glob.glob("$VALID_PATH_GLOB")  # this explicitly types return to list
for _f in files:
    # infinite loop not triggering W4701
    files.append("whatever")

random_list = [1, 2, 3, 4, 5]
for _i in random_list:
    # infinite loop triggering W4701
    random_list.append(6)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:12:4: W4701: Iterated list 'random_list' is being modified inside for loop body, consider iterating through a copy of it instead. (modified-iterating-list)
```

### Expected behavior

```python
************* Module test
test.py:8:4: W4701: Iterated list 'files' is being modified inside for loop body, consider iterating through a copy of it instead. (modified-iterating-list)
test.py:12:4: W4701: Iterated list 'random_list' is being modified inside for loop body, consider iterating through a copy of it instead. (modified-iterating-list)
```

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.12.0 (main, Oct  2 2023, 00:00:00) [GCC 13.2.1 20230918 (Red Hat 13.2.1-3)]
```

### OS / Environment

Fedora 39
